docs/element-definitions.md

```diff
@@ -1,1419 +1,1623 @@
+Type Element matching only character strings. Matches a `string` in JSON.
 
-        If present, the value of the `href` attribute SHOULD be used to resolve
-        the target URI of the transition.
+Because the definition of an Element does not depend on syntax, we might serialize it into JSON:
+```json
 
-        If not set, the parent `resource` element `href` attribute SHOULD be
-        used to resolve the target URI of the transition.
+{
+  "element": "string"
+}
 
-    - `hrefVariables` (Href Variables) - Input parameters.
+```
 
-        Definition of any input URI path segments or URI query parameters for this transition.
+Type Element matching only the character string `"rocket science"`.
 
-        If `href` and `hrefVariables` attributes aren't set, the parent `resource`
-        element `hrefVariables` SHOULD be used to resolve the transition input
-        parameters.
+```json
 
-    - `data` (Data Structure) - Input attributes.
+{
+  "element": "string",
+  "attributes": {
+    "typeAttributes": {
+      "element": "array",
+      "content": [
+        {
+          "element": "string",
+          "content": "fixed"
+        }
+      ]
+    }
+  },
+  "content": "rocket science"
+}
 
-        Definition of any input message-body attribute for this transition.
+```
 
-    - `contentTypes` (Array Element[[String Element][]]) - A collection of content types that MAY be used for the transition.
-- `content` (array)
-    - (Copy) - Transition description's copy text.
-    - (HTTP Transaction) - An instance of transaction example.
-
-        Transaction examples are protocol-specific examples of a REST transaction
-        that was initialized by exercising a transition.
-
-        For the time being this reference document defines only HTTP-specific transaction
-        data structures.
-
-#### Example
-
-```json
-{
-  "element": "transition",
-  "attributes": {
-    "relation": {
-      "element": "string",
-      "content": "update"
-    },
-    "href": {
-      "element": "string",
-      "content": "https://polls.apiblueprint.org/questions/{question_id}"
-    }
-  },
-  "content": []
-}
-```
+---
 
-### API Metadata (Member Element)
+### Array Element
 
-#### Properties
+Type with domain of all finite lists.
 
-- `meta`
-    - `classes` (Array Element)
-        - `content` (array, fixed-type)
-            - ([String Element][])
-                - `content` (enum)
-                    - user (string) - User-specific metadata. Metadata written in the source.
-                    - adapter (string) - Serialization-specific metadata. Metadata provided by adapter.
 
-### Category (Base API Element)
+#### Template
+- `element` - `"array"`
+- `attributes`
+  - `typeAttributes` ([Array](#array-element)[[String](#string-element)])
+    - `fixed` ([String](#string-element)) - Restricts domain to a positionally typed fixed-length list over types in content. This type class is usually called a <i>tuple</i>, <i>sum type</i> or <i>Π-type</i>.
```

Comment:
Let's replace the last sentence with something like:

It further fixes the content of the items given that they declare a value.

Reply:
Further applies the `fixed` type attribute on items declaring a non-sample default.